src/goose/api/batch.clj

```diff
@@ -0,0 +1,27 @@
+(ns goose.api.batch
+  (:require [goose.broker :as b]
+            [goose.client]))
+
+(defn- status-from-counts
+  [{:keys [successful dead]}
+   total]
+  (if (= (+ successful dead) total)
+    "complete"
+    "in-progress"))
+
+(defn- into-batch-map
+  [batch]
+  (let [counts (select-keys batch [:enqueued :retrying :successful :dead])
```

Comment:
question: Different keys of batch i.e [:enqueued :retrying :successful :dead] are mostly redis implementation. Should it even belong in this file?

Reply:
>Different keys of batch i.e [:enqueued :retrying :successful :dead] are mostly redis implementation

Not necessarily, the map that's returned by the API would also have these keys. I've updated the desc of the PR to reflect that.
We're expecting the underlying broker to return a map with these keys.

Reply:
So, irrespective of different implementation of batching by different broker, there will always be [:enqueued :retrying :successful :dead] values given by all broker that implement batching?